vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp

```diff
@@ -939,28 +939,31 @@ void NativeUIManager::DoLayout() {
       assert(false);
       return;
     }
+
+    SetLayoutPropsRecursive(rootTag);
   }
+}
 
-  {
-    SystraceSection s("NativeUIManager::DoLayout::SetLayoutProps");
-    for (auto &tagToYogaNode : m_tagsToYogaNodes) {
-      int64_t tag = tagToYogaNode.first;
-      YGNodeRef yogaNode = tagToYogaNode.second.get();
-
-      if (!YGNodeGetHasNewLayout(yogaNode))
-        continue;
-      YGNodeSetHasNewLayout(yogaNode, false);
-
-      float left = YGNodeLayoutGetLeft(yogaNode);
-      float top = YGNodeLayoutGetTop(yogaNode);
-      float width = YGNodeLayoutGetWidth(yogaNode);
-      float height = YGNodeLayoutGetHeight(yogaNode);
-
-      ShadowNodeBase &shadowNode = static_cast<ShadowNodeBase &>(m_host->GetShadowNodeForTag(tag));
-      auto view = shadowNode.GetView();
-      auto pViewManager = shadowNode.GetViewManager();
-      pViewManager->SetLayoutProps(shadowNode, view, left, top, width, height);
-    }
+void NativeUIManager::SetLayoutPropsRecursive(int64_t tag) {
+  SystraceSection s("NativeUIManager::DoLayout::SetLayoutProps");
```

Comment:
Should this SystraceSection be here, or up above in the initial calling function?

Reply:
Good catch.